### PR TITLE
add socks support to APNS

### PR DESCRIPTION
**Pull request by maleck13**
@matzew Could you take a look at this. Sorry about my IDE removing white space. Will try figure out how to stop it doing that. Do you think this is enough. I can't set the socks proxy at the vm level as it there is a bug in Java that socks proxy properties do not respect the noProxy env var, which means for every socket connection it tries to use the socks proxy event the db!!

### CODE REVIEW

**Comment by maleck13** [comment]
retest this please

**Comment by matzew** [comment]
You expect them as OS/ENV variables ? Or should it be JVM system properties that we pass in with `-Dblah` ?

Not sure I get what you say about the bug, though

**Comment by maleck13** [comment]
@matzew You can set them as system properties. That was how I started using the standalone.xml but then ups couldn't create the persistence unit. Doing a bit more research, I found that when setting these at the jvm level, it will cause all socket connections to try and go through the socks proxy. Normally you can set a ```noProxy``` property to tell it not to proxy things that match the values in the ```noProxy``` property. This is how it works for the http proxy, but this isn't the case for a socks proxy (at least not until a later version of Java)

**Comment by matzew** [comment]
@maleck13 you need to rebase, since I landed some cherry poicks from upstream

**Comment by matzew** [comment]
@maleck13 hey,

I think the upstream project might have usage for that too - do you agree ? 

I am happy to cherry-pick and fix the commit for such an upstream PR

**Comment by maleck13** [comment]
@matzew Yeah that sounds good. Can I leave you to do the merge fhcap voodoo :)

File: admin-ui/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ups-admin-ui",
-  "version": "0.0.0",
+  "version": "1.1.4-3",
```
Comment (matzew) [inline admin-ui/package.json]: @PhilipGough please remove that

**Comment by matzew** [comment]
@PhilipGough nevermind, I just rebased and will fix the compile errors

**Comment by matzew** [comment]
@PhilipGough @maleck13 It compiles now :computer: 

Two things:
* for socks, do we need config for other 'senders' too ? (e.g. the HTTP based FCM/Windows)
* the ping is used for all networks, including the HTTP based ones 

Is that OK :question:

**Comment by matzew** [comment]
Once we are happy with this, @PhilipGough   can you :cherries: pick and send a PR against UPSTREAM (1.1.x-dev) branch ?  Please do not mention RHMAP JIRAs there.

UPS proxy work is bundled in here:
* https://issues.jboss.org/browse/AGPUSH-307

we can use that for the socks proxy part.... 

I think some documetnation, is nice too

File: service/src/main/java/org/jboss/aerogear/unifiedpush/service/impl/health/Ping.java [hunk 2]
```diff
@@ -26,7 +28,12 @@ public static boolean isReachable(String host, int port) {
...
         boolean reachable;
         try {
-            socket = new Socket();
+
+            if (ProxyConfiguration.hasSocks()) {
+                socket = new Socket(ProxyConfiguration.socks());
+            } else {
+                socket = new Socket();
+            }
             socket.connect(new InetSocketAddress(host, port), 2000);
```
Comment (matzew) [inline service/src/main/java/org/jboss/aerogear/unifiedpush/service/impl/health/Ping.java]: @PhilipGough @maleck13 I added this timeout, to make the healthcheck not timepout, if the UPS can not reach the network 

If we want a higher value here, than 2 seconds -> I am totally fine w/ that

Comment (maleck13) [inline service/src/main/java/org/jboss/aerogear/unifiedpush/service/impl/health/Ping.java]: seems reasonable

**Comment by maleck13** [comment]
@matzew They can still set the http proxy env vars which will be picked up for fcm to go through http. Hadn't considered windows. I don't think we need that for this PR but could create a follow on??

**Comment by matzew** [comment]
@maleck13 I am fine with that - I am just concerned, that some community user might be bitten ;-) 

But, we are all OK here, let's 
* close this 
* PR against the 1.1.x-dev branch on upstream

**Pull request closed without merging**